webviz_subsurface/plugins/_well_completions.py

```diff
@@ -0,0 +1,462 @@
+                Output(self.uuid("well_completions_wrapper"), "style"),
+            ],
+            [Input(self.uuid("ensemble_dropdown"), "value"),],
+        )
+        def _render_well_completions(ensemble_name):
+
+            data = json.load(
+                create_ensemble_dataset(
+                    ensemble_name,
+                    self.ens_paths[ensemble_name],
+                    self.compdat_file,
+                    self.zone_layer_mapping_file,
+                    self.well_attributes_file,
+                    self.colors,
+                )
+            )
+            zones = len(data["stratigraphy"])
+            return [
+                webviz_subsurface_components.WellCompletions(
+                    id="well_completions", data=data
+                ),
+                {"padding": "10px", "height": zones * 50 + 180, "min-height": 500},
+            ]
+
+
+@CACHE.memoize(timeout=CACHE.TIMEOUT)
+@webvizstore
+def create_ensemble_dataset(
+    ensemble,
+    ensemble_path,
+    compdat_file,
+    zone_layer_mapping_file,
+    well_attributes_file,
+    colors,
+) -> io.BytesIO:
+    """
+    Creates the well completion data set for the WellCompletions component
+
+    Returns a dictionary on a given format specified here:
+    https://github.com/equinor/webviz-subsurface-components/blob/master/inputSchema/wellCompletions.json
+    """
+
+    df = load_csv(ensemble_paths={ensemble: ensemble_path}, csv_file=compdat_file)
+    qc_compdat(df)
+    layer_zone_mapping = read_zone_layer_mapping(
+        ensemble_path=ensemble_path, zone_layer_mapping_file=zone_layer_mapping_file,
+    )
+    well_attributes = read_well_attributes(
+        ensemble_path=ensemble_path, well_attributes_file=well_attributes_file,
+    )
+
+    time_steps = sorted(df.DATE.unique())
+    realizations = np.asarray(sorted(df.REAL.unique()), dtype=np.int32)
+    layers = np.sort(df.K1.unique())
+
+    if layer_zone_mapping is None:
+        # use layers as zones
+        layer_zone_mapping = {layer: f"Layer{layer}" for layer in layers}
+
+    df["ZONE"] = df.K1.map(layer_zone_mapping)
+
+    result = {}
+    result["version"] = "1.0.0"
+    result["stratigraphy"] = extract_stratigraphy(layer_zone_mapping, colors)
+    result["timeSteps"] = time_steps
+
+    zone_names = [a["name"] for a in result["stratigraphy"]]
+    result["wells"] = extract_wells(
+        df, zone_names, time_steps, realizations, well_attributes
+    )
+
+    return io.BytesIO(json.dumps(result).encode())
+
+
+def qc_compdat(compdat):
+    """
+    QCs that the compdat data has the required format
+    """
+    needed_columns = ["WELL", "K1", "OP/SH", "KH"]
+    for column in needed_columns:
+        if column not in compdat:
+            raise ValueError(
+                f"""
+                Column {column} not found in compdat dataframe.
+                This should not occur unless there has been changes to ecl2df.
+                """
```

Comment:
```suggestion
                f"Column {column} not found in compdat dataframe. "
                "This should not occur unless there has been changes to ecl2df."
```
Otherwise I think the error message would contain the extra white space paddings.